core/src/main/java/brooklyn/enricher/basic/Transformer.java

```diff
@@ -102,13 +95,65 @@ public void setEntity(EntityLocal entity) {
         }
     }
 
+    /** returns a function for transformation, for immediate use only (not for caching, as it may change) */
+    @SuppressWarnings("unchecked")
+    protected Function<SensorEvent<T>, U> getTransformation() {
```

Comment:
it wouldn't change, but this design makes it easier to support changing config in future. if it's an efficiency hole we can switch to populate this at start.  will add a comment to this effect.